examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,882 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# `ConditionalDimension` tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial explains how to create equations that only get executed under given conditions.\n",
+    "\n",
+    "As in all our tutorials, we begin by specifying some parameters.\n",
+    "For our examples we will use `Function`s defined on 2D grids and their data values are going to be updated according to prescribed conditions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 1., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 1., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 1., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 1., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 1., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "from devito import Dimension, Function, Grid\n",
+    "import numpy as np\n",
+    "\n",
+    "# We define a 10x10 grid, dimensions are x, y\n",
+    "shape = (10, 10)\n",
+    "grid = Grid(shape = shape)\n",
+    "x, y = grid.dimensions\n",
+    "\n",
+    "# Define function 𝑓. We will populate f's data with ones on its diagonal.\n",
+    "\n",
+    "f = Function(name='f', grid=grid)\n",
+    "f.data[:, :] = np.eye(10)\n",
+    "f.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now lets construct a simple `Operator` to increase the values of $ f $ by 1."
```

Comment:
We begin by constructing an Operator that increases the values of `f`, a `Function` defined on a two-dimensional `Grid`, by one.

Reply:
Fixed